gems/pending/spec/postgres_ha_admin/failover_monitor_spec.rb

```diff
@@ -0,0 +1,141 @@
+        failover_monitor.monitor
+      end
+
+      it "does not stop evm server and does not execute failover" do
+        expect(failover_db).to receive(:update_failover_yml)
+        expect(linux_admin).not_to receive(:stop)
+        expect(linux_admin).not_to receive(:restart)
+        expect(failover_monitor).not_to receive(:execute_failover)
+
+        failover_monitor.monitor
+      end
+    end
+
+    context "primary database is not accessable" do
+      before do
+        allow(PG::Connection).to receive(:open).and_return(nil, connection, connection)
+      end
+
+      it "stop evm server(if it is running) before failover attempt" do
+        expect(linux_admin).to receive(:stop).ordered
+        expect(failover_monitor).to receive(:execute_failover).ordered
+
+        failover_monitor.monitor
+      end
+
+      it "does not update 'database.yml' and 'failover_databases.yml' if all standby DBs are in recovery mode" do
+        expect(failover_db).to receive(:active_databases).and_return(active_databases_list)
+        expect(failover_db).not_to receive(:update_database_yml)
+        expect(db_yml).not_to receive(:update_database_yml)
+        expect(linux_admin).not_to receive(:restart)
+
+        expect(failover_monitor).to receive(:stop_evmserverd).ordered
+        expect(PostgresAdmin).to receive(:database_in_recovery?).and_return(true, true, true).ordered
+
+        stub_const("PostgresHaAdmin::FailoverMonitor::FAILOVER_ATTEMPTS", 1)
+        stub_const("PostgresHaAdmin::FailoverMonitor::FAILOVER_CHECK_FREQUENCY", 1)
+        failover_monitor.monitor
+      end
+
+      it "does not update 'database.yml' and 'failover_databases.yml' if there is no master database avaiable" do
+        expect(failover_db).to receive(:active_databases).and_return(active_databases_list)
+        expect(failover_db).not_to receive(:update_database_yml)
+        expect(db_yml).not_to receive(:update_database_yml)
+        expect(linux_admin).not_to receive(:restart)
+
+        expect(PostgresAdmin).to receive(:database_in_recovery?).and_return(false, false, false)
+        expect(linux_admin).to receive(:stop).ordered
+        expect(failover_db).to receive(:host_is_repmgr_primary?).and_return(false, false, false).ordered
+
+        stub_const("PostgresHaAdmin::FailoverMonitor::FAILOVER_ATTEMPTS", 1)
+        stub_const("PostgresHaAdmin::FailoverMonitor::FAILOVER_CHECK_FREQUENCY", 1)
+        failover_monitor.monitor
+      end
+
+      it "updates 'database.yml' and 'failover_databases.yml' and restart evm server if new primary db available" do
+        allow(PostgresAdmin).to receive(:database_in_recovery?).and_return(false)
+        allow(failover_db).to receive(:host_is_repmgr_primary?).and_return(guery_repmanager_result)
+        allow(failover_db).to receive(:active_databases).and_return(active_databases_list)
+        allow(failover_db).to receive(:update_failover_yml).with(connection)
```

Comment:
OK, grouping common expectation will make it more readable.

While using ```expect``` in previous tests (not executed failover) i am making sure that stubbed method would be called more that 2 times and all supplied data will be enumerated.

Last expectation(successful failover) may find available database somewhere in the middle of list and not all data would be enumerated. Using ```expect``` here will require to request specific number of iteration, I think may make code less readable since we just need to supplying data